tls/s2n_x509_validator.c

```diff
@@ -486,6 +495,71 @@ static S2N_RESULT s2n_x509_validator_process_cert_chain(struct s2n_x509_validato
     return S2N_RESULT_OK;
 }
 
+#ifdef S2N_LIBCRYPTO_SUPPORTS_FLAG_NO_CHECK_TIME
+static S2N_RESULT s2n_x509_validator_set_no_check_time_flag(struct s2n_x509_validator *validator)
+{
+    RESULT_ENSURE_REF(validator);
+    RESULT_ENSURE_REF(validator->store_ctx);
+
+    X509_VERIFY_PARAM *param = X509_STORE_CTX_get0_param(validator->store_ctx);
+    RESULT_ENSURE_REF(param);
+
+    RESULT_GUARD_OSSL(X509_VERIFY_PARAM_set_flags(param, X509_V_FLAG_NO_CHECK_TIME),
+            S2N_ERR_INTERNAL_LIBCRYPTO_ERROR);
+
+    return S2N_RESULT_OK;
+}
+#else
+static S2N_RESULT s2n_x509_validator_set_no_check_time_flag(struct s2n_x509_validator *validator)
+{
+    RESULT_BAIL(S2N_ERR_UNIMPLEMENTED);
+}
+#endif
+
+int s2n_disable_validity_period_validation_ossl_verify_callback(int default_ossl_ret, X509_STORE_CTX *ctx)
+{
+    int err = X509_STORE_CTX_get_error(ctx);
+    switch (err) {
+        case X509_V_ERR_CERT_NOT_YET_VALID:
+        case X509_V_ERR_CERT_HAS_EXPIRED:
+            return 1;
+        default:
+            /* If CRL validation is enabled, the CRL verify callback is set on the X509_STORE_CTX.
+             * If the validity period verify callback is set on the CTX, this will override the
+             * CRL verify callback. For simplicity, the CRL verify callback is called in addition
+             * to the validity period verify callback, ensuring both features can be enabled
+             * simultaneously.
+             */
+            return s2n_crl_ossl_verify_callback(default_ossl_ret, ctx);
+    }
```

Comment:
This is pretty confusing. I think you're delegating to s2n_crl_ossl_verify_callback if we don't want to override the default_ossl_ret?

It might be clearer if you call this after the switch statement. Like:
```suggestion
    int err = X509_STORE_CTX_get_error(ctx);
    switch (err) {
        case X509_V_ERR_CERT_NOT_YET_VALID:
        case X509_V_ERR_CERT_HAS_EXPIRED:
            return 1;
    }
    /* If CRL validation is enabled, the CRL verify callback is set on the X509_STORE_CTX.
     * If the validity period verify callback is set on the CTX, this will override the
     * CRL verify callback. Manually call the CRL verify callback so that both features
     * can be enabled at the same time.
     */
    return s2n_crl_ossl_verify_callback(default_ossl_ret, ctx);
```